src/kakarot/account.cairo

```diff
@@ -0,0 +1,337 @@
+    // @notice Iterate through the storage dict and update the Starknet storage
+    func commit{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        self: Summary*, starknet_address: felt
+    ) {
+        alloc_locals;
+
+        let (registered_starknet_account) = Accounts.get_starknet_address(self.address);
+        let starknet_account_exists = is_not_zero(registered_starknet_account);
+
+        // Case new Account
+        if (starknet_account_exists == 0) {
+            // Deploy accounts
+            let (class_hash) = contract_account_class_hash.read();
+            Accounts.create(class_hash, self.address);
+            // Write bytecode
+            IContractAccount.write_bytecode(starknet_address, self.code_len, self.code);
+            // Set nonce
+            IContractAccount.set_nonce(starknet_address, self.nonce);
+            // Save storages
+            Internals._save_storage(starknet_address, self.storage_start, self.storage);
+            return ();
+        }
+
+        // Case SELFDESTRUCT
+        if (self.selfdestruct != 0) {
+            // SELFDESTRUCT
+            // TODO: clean also the storage
+            let (local erase_data: felt*) = alloc();
+            Helpers.fill(self.code_len, erase_data, 0);
+            IContractAccount.write_bytecode(
+                contract_address=starknet_address, bytecode_len=self.code_len, bytecode=erase_data
+            );
+            return ();
+        }
+
+        // Case EOA
+        // TODO: use supports interface instead of the bytecode_len proxy
+        let (bytecode_len) = Accounts.get_bytecode_len(self.address);
+        if (bytecode_len == 0) {
+            return ();
+        }
+
+        // Set nonce
+        IContractAccount.set_nonce(starknet_address, self.nonce);
+        // Save storages
+        Internals._save_storage(starknet_address, self.storage_start, self.storage);
+
+        return ();
+    }
+
+    // @notice fetch an account from Starknet
+    // @dev An non deployed account is just an empty account.
+    // @param address the pointer to the Address
+    // @return the account populated with Starknet data
+    func fetch_or_create{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        address: model.Address*
+    ) -> model.Account* {
+        alloc_locals;
+        let (local registered_starknet_account) = Accounts.get_starknet_address(address.evm);
+        let (bytecode_len, bytecode) = Accounts.get_bytecode(address.evm);
+        let starknet_account_exists = is_not_zero(registered_starknet_account);
+
+        // Case touching a non deployed account
+        if (starknet_account_exists == 0) {
+            let account = Account.init(
+                address=address.evm, code_len=bytecode_len, code=bytecode, nonce=0
+            );
+            return account;
+        }
+
+        // Case EOA
+        // TODO: use supports interface instead of the bytecode_len proxy
+        if (bytecode_len == 0) {
```

Comment:
what if it's a CA that has no bytecode?

Reply:
we use the registry here, what do you mean?

Reply:
Maybe I misunderstood, it's written, case: EOA
But I asked if when we do:
fetch bytecode,
if we fetch the bytecode of an already deployed contract account, that has no bytecode, then bytecode_len == 0

Reply:
so we're rekt, that's the point of the comment, that it's not possible to know and we're missing an entrypoint. but in any case a CA with no bytecode cannot bring any change to the state, so its nonce has no impact